internal/incoming/odfi/download.go

```diff
@@ -123,7 +134,7 @@ func (dl *downloaderImpl) setup(agent upload.Agent) (*downloadedFiles, error) {
 	}, nil
 }
 
-func (dl *downloaderImpl) CopyFilesFromRemote(agent upload.Agent, shard *service.Shard) (*downloadedFiles, error) {
+func (dl *downloaderImpl) CopyFilesFromRemote(ctx context.Context, agent upload.Agent, shard *service.Shard) (*downloadedFiles, error) {
```

Comment:
Is it possible to avoid passing the context around so much?

Reply:
Not really. Contexts are created for each request/message through the system and store arbitrary information like tracing info. 

If we had some sort of `*Request` struct the context could be in there and hidden a bit better.